utils/crud/backingStore.go

```diff
@@ -0,0 +1,119 @@
+package crud
+
+var _ Store = &BackingStore{}
+
+// BackingStore wraps another store that may have Connect/Close methods that
+// need to be called.
+// - Connect is called before a method when the connection is closed.
+// - Close is called after each method when AutoClose is true (default).
+type BackingStore struct {
+	AutoClose    bool
```

Comment:
I think `AutoClose` shouldn't be exposed, and should be configured only once at construction time. We can either add a boolean to the `NewBackingStore` function, or have a functional options pattern (but here it may be too much as we only have one parameter? 🤔 ). WDYT?

Reply:
I made it an exported field because I didn't expect anyone to use it often honestly, and wanted AutoClose to be defaulted to True. This seemed the most straightforward way to let people change that if they needed more control.

Is there a problem with the exported field that I am overlooking? Just to limit exports?